### ISSUE TITLE
Add support for Bitbucket OIDC provider and Bitbucket OIDC role

### ISSUE THREAD

**Issue by sirantd** [terraform-aws-modules/terraform-aws-iam#336]
Is your request related to a new offering from AWS?
Is this functionality available in the [AWS provider for Terraform](https://registry.terraform.io/providers/hashicorp/aws/latest/docs)? See [CHANGELOG.md](https://github.com/hashicorp/terraform-provider-aws/blob/main/CHANGELOG.md), too.
  - Yes ✅: please list the AWS provider version which introduced this functionality

## Is your request related to a problem? Please describe.
We have inbuilt support for creating GitHub OIDC provider but it can be nice if can use the same module for Bitbucket OIDC provider

## Describe the solution you'd like.
Create two additional submodules for provider and role

#337

**Comment by sirantd** [comment]
PR #337 was created to solve issue

**Comment by wiseelf** [comment]
would be nice to have it, because I use self-written module for that.

**Comment by sirantd** [comment]
Hi @bryantbiggs, could you please check if it's something that would be nice to have as part of the module (with further adding Gitlab OIDC the same way) or it's not gets merged for other reasons (bad PR).

**Comment by bryantbiggs** [comment]
I think it would be nice to have, but I am afraid the pattern we started is not one that is sustainable so I'm hesitant to continue expanding that pattern. For example, [here is a fork](https://github.com/clowdhaus/terraform-aws-iam/tree/master) of this project I have been working on that would reduce the amount of sub-modules we have here by making them more generic. It covers all of the same functionality offered by this module, but also supports the various different configurations. You can see with the OIDC provider and OIDC role, users can use any source control provider (or any OIDC authority in general).

**Comment by sirantd** [comment]
Thank you for explanation, I see it's much cleaner in the fork.
I understand that it's impossible to have back compatibility with the current module, so what is the plan? Depricate this module (and submodules) from specific version and merging your fork in or have standalone fork in parallel?
